app/javascript/dashboard/components-next/button/Button.vue

```diff
@@ -11,36 +11,60 @@ const props = defineProps({
   },
   variant: {
     type: String,
-    default: 'solid',
+    default: null,
     validator: value =>
-      ['solid', 'outline', 'faded', 'link', 'ghost'].includes(value),
+      ['solid', 'outline', 'faded', 'link', 'ghost'].includes(value) ||
+      value === null,
   },
   color: {
     type: String,
-    default: 'blue',
+    default: null,
     validator: value =>
-      ['blue', 'ruby', 'amber', 'slate', 'teal'].includes(value),
+      ['blue', 'ruby', 'amber', 'slate', 'teal'].includes(value) ||
+      value === null,
   },
   size: {
     type: String,
-    default: 'md',
-    validator: value => ['xs', 'sm', 'md', 'lg'].includes(value),
-  },
-  icon: {
-    type: String,
-    default: '',
-  },
-  trailingIcon: {
-    type: Boolean,
-    default: false,
-  },
-  isLoading: {
-    type: Boolean,
-    default: false,
+    default: null,
+    validator: value =>
+      ['xs', 'sm', 'md', 'lg'].includes(value) || value === null,
   },
+  icon: { type: [String, Object, Function], default: '' },
+  trailingIcon: { type: Boolean, default: false },
+  isLoading: { type: Boolean, default: false },
 });
 
 const slots = useSlots();
+const attrs = useAttrs();
+
+const computedVariant = computed(() => {
+  if (props.variant) return props.variant;
+  if (attrs.solid !== undefined) return 'solid';
```

Comment:
```suggestion
  if (attrs.solid) return 'solid';
```

Is it mandatory to check this with `undefined`? Shouldn’t `attrs.solid` suffice?

Reply:
Fixed it here: https://github.com/chatwoot/chatwoot/pull/10473